pkg/controller/cincinnati/cincinnati_controller.go

```diff
@@ -231,22 +231,30 @@ func (r *ReconcileCincinnati) ensureAdditionalTrustedCA(ctx context.Context, req
 		return err
 	}
 
-	localCM := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      nameAdditionalTrustedCA(instance),
-			Namespace: instance.Namespace,
-		},
-		Data: sourceCM.Data,
-	}
+	if _, ok := sourceCM.Data[NameCertConfigMapKey]; ok {
+		localCM := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      nameAdditionalTrustedCA(instance),
+				Namespace: instance.Namespace,
+			},
+			Data: sourceCM.Data,
+		}
 
-	// Set Cincinnati instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, localCM, r.scheme); err != nil {
-		return err
-	}
+		// Set Cincinnati instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, localCM, r.scheme); err != nil {
+			return err
+		}
 
-	if err := r.ensureConfigMap(ctx, reqLogger, localCM); err != nil {
-		handleErr(reqLogger, &instance.Status, "EnsureConfigMapFailed", err)
-		return err
+		if err := r.ensureConfigMap(ctx, reqLogger, localCM); err != nil {
+			handleErr(reqLogger, &instance.Status, "EnsureConfigMapFailed", err)
+			return err
+		}
+		// Mount in ConfigMap data from the cincinnati-registry key
+		ExternalCACert := true
+		resources.graphBuilderContainer = resources.newGraphBuilderContainer(instance, r.operandImage, ExternalCACert)
+		resources.deployment = resources.newDeployment(instance, ExternalCACert)
```

Comment:
Change the private variable to start with lowercase: `externalCACert`

Nit: WDYT about making the externalCACert a property of the ReconcileCincinnati struct, and initializing it to true in newReconciler() function or the beginning of the Reconcile? That way, you can create the newKubeResources() with externalCACert and not have to selectively override some of the resources.

Reply:
I'm essentially copying around the same value, so I agree stylistically.   The second part of your comment: `not have to selectively override some of the resources`, would imply that once `ExternalCACert` changes to `true`, we need to trigger a Reconcile event.  That would make for a trade-off in that we won't be progressing forward because we would have to `return` after the change.  That might not be a big deal though, thoughts?

Reply:
Not sure I totally understand, but I'd check for r.externalCACert in ensureAdditionalCACert() and return immediately if it's set to false. I don't think we need the ConfigMap created if externalCACert is false.

Reply:
Right, I think we're saying similar things. Once `ensureAdditionalCACert()` notices that `r.externalCACert` has changed, it would have to wait until the next `Reconcile()` call to pick up the correct kubeResources and apply them.  I'm just curious if anyone thinks that's an anti-pattern as opposed to fixing and applying the kuberesources on the current `Reconcile()` call.

Reply:
Ok, ignore my comment. I didn't realize it was set to true only in the case when the configMap key was found.